drivers/mtd/mtd.c

```diff
@@ -167,11 +167,16 @@ int mtd_write_page(mtd_dev_t *mtd, const void *data, uint32_t page,
         return res;
     }
 
+    /* prevent heap overflow of work buffer */
+    if (count + offset > sector_size) {
```

Comment:
The offset inside the sector should not matter as long as it is < `sector_size`.

Reply:
But below this line is `memcpy(work + offset, data, count);`. If count + offset is bigger than sector size this will overflow.

Reply:
Yes you are right. But we should check this *before* erasing the page.

Reply:
Moved the check before any reading / erasing is done.